app/_src/quickstart/kubernetes-demo.md

```diff
@@ -0,0 +1,147 @@
+
+Install Kuma control plane with Helm by executing:
+
+```sh
+helm repo add kuma https://kumahq.github.io/charts
+helm repo update
+helm install --create-namespace --namespace {{site.mesh_namespace}} {{ site.mesh_helm_install_name }} {{ site.mesh_helm_repo }}
+```
+
+## Deploy demo application
+
+1.  Deploy the application
+    ```sh
+    kubectl apply -f https://raw.githubusercontent.com/kumahq/kuma-counter-demo/master/demo.yaml
+    ```
+
+2.  Port-forward the service to the namespace on port 5000:
+
+    ```sh
+    kubectl port-forward svc/demo-app -n kuma-demo 5000:5000
+    ```
+
+3.  In a browser, go to [127.0.0.1:5000](http://127.0.0.1:5000) and increment the counter.
+
+The demo app includes the `kuma.io/sidecar-injection` label enabled on the `kuma-demo` namespace.
+
+```yaml
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: kuma-demo
+  labels:
+    kuma.io/sidecar-injection: enabled
+```
+
+This means that {{site.mesh_product_name}} [already knows](/docs/{{ page.version }}/production/dp-config/dpp-on-kubernetes/) that it needs to automatically inject a sidecar proxy to every Kubernetes pod in the `kuma-demo` namespace.
+
+## Explore GUI
+
+You can view the sidecar proxies that are connected to the {{site.mesh_product_name}} control plane.
+
+{{site.mesh_product_name}} ships with a **read-only** GUI that you can use to retrieve {{site.mesh_product_name}} resources. By default, the GUI listens on the API port which defaults to `5681`.
+
+To access {{site.mesh_product_name}} we need to first port-forward the API service with:
+
+```sh
+kubectl port-forward svc/{{site.mesh_cp_name}} -n {{site.mesh_namespace}} 5681:5681
+```
+
+And then navigate to [127.0.0.1:5681/gui](http://127.0.0.1:5681/gui) to see the GUI.
+
+## Introduce zero-trust security
+
+By default, the network is insecure and not encrypted. We can change this with {{site.mesh_product_name}} by enabling the [Mutual TLS](/docs/{{ page.version }}/policies/mutual-tls/) policy to provision a Certificate Authority (CA) that will automatically assign TLS certificates to our services (more specifically to the injected data plane proxies running alongside the services).
+
+We can enable Mutual TLS with a `builtin` CA backend by executing:
+
+```sh
+echo "apiVersion: kuma.io/v1alpha1
+kind: Mesh
+metadata:
+  name: default
+spec:
+  mtls:
+    enabledBackend: ca-1
+    backends:
+    - name: ca-1
+      type: builtin" | kubectl apply -f -
+```
+
+The traffic is now encrypted with mTLS. However, every service can reach any other service.
+
+We can then restrict the traffic by default by executing:
+
+```sh
+echo "
+apiVersion: kuma.io/v1alpha1
+kind: MeshTrafficPermission
+metadata:
+  namespace: {{site.mesh_namespace}}
+  name: deny-all
+spec:
+  targetRef:
+    kind: Mesh
+  from:
+    - targetRef:
+        kind: Mesh
+      default:
+        action: Deny" | kubectl apply -f -
+```
+
+At this point, the demo application should not function, because we blocked the traffic.
```

Comment:
Again, just one of those "you never know". I kinda figured that someone might be like "How do I know its not working? (refresh page, refresh page....)", so maybe we add "you can verify this by clicking the increment button again and seeing the error message in the browser"